### PR TITLE
[AIRFLOW-XXXX] Expose SQLAlchemy's connect_args and make it configurable

### PR DESCRIPTION
**Pull request by ZxMYS**
#6226 # Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
I didn't create JIRA ticket

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:
In our use case we need to configure SQLAlchemy's connect_args (e.g. we want to pass ssl.check_hostname=False to PyMySQL), and Airflow should expose this option and make it configurable.

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
test_configure_orm_with_default_values, test_sql_alchemy_connect_args and test_sql_alchemy_invalid_connect_args in the new tests/test_sqlalchemy_config.py.

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.

### CODE REVIEW

**Comment by kaxil** [comment]
This can already be done using `db = create_engine('postgresql://scott:tiger@localhost/test?argument1=foo&argument2=bar')`

Do you have any use case for this?

**Comment by ZxMYS** [comment]
@kaxil  user can not pass in a dict in connection string, which is needed if we want to set `ssl = { check_hostname: False }` for pymysql.
Passing arguments in connection string only works if SqlAlchemy recognizes the arguments and parses them properly - but unfortunately SqlAlchemy doesn't recognize some of the newer or more complex arguments, e.g. it supports `ssl_ca`, but not `ssl_check_hostname`.

**Comment by aoen** [comment]
Can you add your reply to Kaxil as a comment in the code for posterity?

I think we may want a test for this too. It would show how to use it, plus of course add test coverage : ). airflow/tests/test_logging_config.py is an example of how to do this.

**Comment by kaxil** [comment]
> Can you add your reply to Kaxil as a comment in the code for posterity?
> 
> I think we may want a test for this too. It would show how to use it, plus of course add test coverage : ). airflow/tests/test_logging_config.py is an example of how to do this.

Agree with Dan

**Comment by ZxMYS** [comment]
ok, I updated description in default_airflow.cfg and added some simple unit test cases. Let me know if they are enough :)

**Comment by aoen** [comment]
Linting is failing @ZxMYS https://travis-ci.org/apache/airflow/jobs/605815560?utm_medium=notification&utm_source=github_status

File: tests/test_sqlalchemy_config.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+# under the License.
+
+import unittest
+
+from sqlalchemy.pool import NullPool
+from airflow import settings
+from tests.compat import patch
+from tests.test_utils.config import conf_vars
+
+
```
Comment (kaxil) [inline tests/test_sqlalchemy_config.py]:
```suggestion
```

File: tests/test_sqlalchemy_config.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import unittest
+
+from sqlalchemy.pool import NullPool
+from airflow import settings
```
Comment (kaxil) [inline tests/test_sqlalchemy_config.py:24]:
```suggestion

from airflow import settings
```

**Comment by ZxMYS** [comment]
Lint issues fixed! I did run `./breeze  --static-check pylint -- --files  tests/test_sqlalchemy_config.py`, not sure why it didn't catch the issues appeared on CI

**Comment by aoen** [comment]
LGTM

**Comment by bhavaniravi** [comment]
Is this feature available on 1.10.10?

**Pull request merged**